cross-modal-search/flows/flow-query.yml

```diff
@@ -1,43 +1,42 @@
-jtype: Flow
-version: '1'
+jtype: Flow                                       # This file defines the query flow which is used for searching in the indexed documents
+version: '1'                                      # The query flow is very similar to the index flow - only the differences are explained here
 with:
   prefetch: 10
-  port_expose: 45678
+  port_expose: $JINA_PORT
   workspace: $JINA_WORKSPACE
 pods:
-  - name: loader                          # load query image
-    uses: pods/image-load.yml
-    shards: $JINA_PARALLEL
+  - name: loader                                  # Again, we start two paths in the flow - here we start the image path
+    uses: ImageReader
+    py_modules: 'flows/executors.py'
     read_only: true
-    needs: gateway
-  - name: image_encoder                   # encode query image into embeddings with CLIP model
-    polling: any
-    uses: 'jinahub://CLIPImageEncoder'
+    needs: [gateway]
+  - name: image_encoder                           # Now, encode the images and compute the embeddings
+    uses: 'jinahub+docker://CLIPImageEncoder'
     volumes: $HOME/.cache/clip:/root/.cache/clip
-    shards: $JINA_PARALLEL
     timeout_ready: 600000
     read_only: true
-    needs: loader
-  - name: text_indexer                    # index query text
-    polling: all
-    uses: pods/index-text.yml
-    shards: $JINA_SHARDS
-  - name: text_encoder                    # encode query text into embeddings with CLIP model
-    uses: pods/clip/text-encoder.yml
-    shards: $JINA_PARALLEL
-    timeout_ready: 600000
+  - name: text_indexer                            # Now, we use the text indexer in the image path - This is how we achieve the cross-modality here
+    uses: 'docker://simple_indexer'               # The text indexer has indexed all text documents and stored them on disk.
```

Comment:
Is there a reason why this is not consistent with the one in `flow-index.yml` (i.e. `jinahub+docker://SimpleIndexer`)?

Reply:
Actually, I am getting the following error:
`ext_indexer@18362[E]:can not find remote image: simple_indexer
   text_indexer@18362[E]:BadImageNameError('image: simple_indexer can not be found local & remote.') during <class 'jina.peapods.runtimes.container.ContainerRuntime'> initialization
 add "--quiet-error" to suppress the exception details`

Reply:
+1

Reply:
Yes that is a bug, my bad